### PR TITLE
document private vars to alert users if they're shadowing them

### PR DESCRIPTION
**Pull request by staeke**
This is to avoid bugs that can be caught by Typescript

### CODE REVIEW

File: src/preact.d.ts [hunk 1]
```diff
@@ -99,6 +99,17 @@ declare namespace preact {
 		forceUpdate(callback?: () => void): void;
 
 		abstract render(props?: RenderableProps<P>, state?: Readonly<S>, context?: any): ComponentChild;
+
+		// Add these variables to avoid descendants shadowing them
+		private __key;
```
Comment (developit) [inline src/preact.d.ts]: In the Preact codebase, any properties with a leading underscore get minified using the mapping from here:
https://github.com/developit/preact/blob/master/config/properties.json

In addition, `next*` and `prev*` are also minified.  We might as well add them all if we're looking to warn folks.

Comment (staeke) [reply src/preact.d.ts]: Good catch. I'll add them as well

File: src/preact.d.ts [hunk 2]
```diff
@@ -99,6 +99,32 @@ declare namespace preact {
...
+		private __h;
+		private __k;
+		private __r;
+		private __n;
+		private __b;
+		private __c;
+		private __p;
+		private __s;
+		private __u;
+		private _componentConstructor;
```
Comment (developit) [inline src/preact.d.ts]: nit: These last three are only ever used on HTML Elements.

**Comment by developit** [comment]
@staeke want me to merge and remove the last 3?

**Comment by staeke** [comment]
Sure. Pls

**Comment by ForsakenHarmony** [comment]
Released in v8.5.0, sorry it took this long

**Pull request merged**